File: src/adapter/festa/fantasia/ExemploFestaFantasia.java
package adapter.festa.fantasia;

public class ExemploFestaFantasia {

	public static void main(String[] args) {
			
		FestaDeHalloween festa = new FestaDeHalloween();
		
		// Primeiros participantes a chegar
		
		Pessoa joao = new Pessoa("João", "Refrigerante");
		Pessoa lucas = new Pessoa("Lucas", "Suco de Laranja");
		Pessoa gabriel = new Pessoa("Gabriel", "Cerveja");
		Pessoa fabio = new Pessoa("Fabio", "Refrigerante");
		
		festa.entrar(new RoupaDeFantasma(joao));
		festa.entrar(new RoupaDeVampiro(lucas));
		festa.entrar(new RoupaDeCaveira(gabriel));
		festa.entrar(new RoupaDeVampiro(fabio));
		
		festa.exibirParticipantesQueBebem("Refrigerante");
		// Saída: João e Fábio
		
		festa.exibirTodosOsParticipantes();
	}

}

File: src/adapter/festa/fantasia/ExemploFestaFantasia.kt
package adapter.festa.fantasia

object ExemploFestaFantasia {

    @JvmStatic
    fun main(args: Array<String>) {

        val festa = FestaDeHalloween()

        // Primeiros participantes a chegar

        val joao = Pessoa("João", "Refrigerante")
        val lucas = Pessoa("Lucas", "Suco de Laranja")
        val gabriel = Pessoa("Gabriel", "Cerveja")
        val fabio = Pessoa("Fabio", "Refrigerante")

        festa.entrar(RoupaDeFantasma(joao))
        festa.entrar(RoupaDeVampiro(lucas))
        festa.entrar(RoupaDeCaveira(gabriel))
        festa.entrar(RoupaDeVampiro(fabio))

        festa.exibirParticipantesQueBebem("Refrigerante")
        // Saída: João e Fábio

        festa.exibirTodosOsParticipantes()
    }

}
